router/middleware/session/team.go

```diff
@@ -1,21 +1,93 @@
 package session
 
 import (
+	"github.com/drone/drone/cache"
+	"github.com/drone/drone/model"
+
+	log "github.com/Sirupsen/logrus"
 	"github.com/gin-gonic/gin"
 )
 
+func TeamPerm(c *gin.Context) *model.Perm {
+	v, ok := c.Get("teamperm")
+	if !ok {
+		return nil
+	}
+	u, ok := v.(*model.Perm)
+	if !ok {
+		return nil
+	}
+	return u
+}
+
+func SetTeamPerm() gin.HandlerFunc {
```

Comment:
I'm wondering if we need this function. The permissions are only ever viewed in a single place, MustTeamAdmin, so attaching more data to the context probably isn't necessary. But open to hear what everyone else thinks.

As a side note, I think if we were to use this middleware, it would need to be registered with the handler.